src/Idris/Driver.idr

```diff
@@ -182,7 +182,9 @@ stMain cgs opts
                                   displayErrors res
                                   pure res
 
-                 doRepl <- postOptions result opts
```

Comment:
Prior to this change, `doRepl` would become `False` and we would
wound up taking the `else` branch below. Which means that in
addition to displaying the error & exiting we would `showTimeRecord`.

Could we *not* bypass that if-then-else so that we keep executing the
same block of code rather than duplicating the logic? I guess something
like

```
doRepl <- catch (postOptions result opts)
                (\ err => pure False)
```

ought to work. If it does not then we may need to register that `err`
so that it is noticed by `errorLine` down below. But I would expect
that to already be done at the point of emission.

Reply:
Oops yes I'll add `showTimeRecord` into there

errorLine is a little more bare than you could be thinking, it just stores the line number where an error occurs and is only updated in one or two places (via updateErrorLine). There isn't actually error displaying happening in the original code, just an `exitWith (ExitFailure 1)` if errorLine is Just. My guess was that error reporting was expected to happen before that point which is why I did it at `doRepl`.

There are a couple options to explore if we want to report errors, such as bundling the error with doRepl to print later when we exit:
```idris
                 (doRepl,err) <- catch (pure (!(postOptions result opts), Nothing))
                                       (\err => pure (False, Just err))
```
Or to change errorLine to store the error in-addition-to/instead-of just the line.

Both of these are more involved than just repeating the call to showTimeRecord in the catch though. Thoughts?

Reply:
I guess we already have a duplicated `showTimeRecord` after the call to
`repl` in a different branch so we may as well do the same here.